projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -73,6 +73,20 @@ public function reregister_menu_items() {
 			if ( $nudge ) {
 				parent::add_upsell_nudge( $nudge );
 			}
+
+			/**
+			 * Whether to show the WordPress.com Inbox menu under Upgrades menu.
+			 *
+			 * @use add_filter( 'jetpack_show_wpcom_inbox_menu', '__return_true' );
+			 * @module masterbar
+			 *
+			 * @since 9.7.0
+			 *
+			 * @param bool $jetpack_show_wpcom_inbox_menu Load the WordPress.com Inbox menu item. Default to false.
+			 */
+			if ( apply_filters( 'jetpack_show_wpcom_inbox_menu', false ) ) {
```

Comment:
I believe that you can move the filter checks/gates into to the main body of `\Automattic\Jetpack\Dashboard_Customizations\Admin_Menu::add_inbox_menu` itself like this:

```php

	/**
	 * Adds Inbox menu.
	 */
	public function add_inbox_menu() {
                // Checks for the filter here ...
		if ( apply_filters( 'jetpack_show_wpcom_inbox_menu', false ) ) {
			add_menu_page( __( 'Inbox', 'jetpack' ), __( 'Inbox', 'jetpack' ), 'edit_posts', 'https://wordpress.com/inbox/' . $this->domain, null, 'dashicons-email', '4.64424' );
		}
	}
```

This way, we only need to refactor it there ONCE.

Reply:
I moved it there because tests needed to add the filters to work, I'll do it anyway 🙂